blark/transform.py

```diff
@@ -614,30 +614,43 @@ class Variable(Expression):
     ...
 
 
+@dataclass
 @_rule_handler(
     "indirection_type",
     "pointer_type",
 )
-class IndirectionType(Enum):
+class IndirectionType(Literal):
     """Indirect access through a pointer or reference."""
-    none = enum.auto()
-    pointer = enum.auto()
-    reference = enum.auto()
+    pointer_depth: int
+    reference: bool
+    meta: Optional[Meta] = meta_field()
 
     @staticmethod
-    def from_lark(token: Optional[lark.Token]) -> IndirectionType:
-        return {
-            "NONE": IndirectionType.none,
-            "POINTER TO": IndirectionType.pointer,
-            "REFERENCE TO": IndirectionType.reference,
-        }[str(token).upper()]
+    def from_lark(*tokens: Tuple[lark.Token]) -> IndirectionType:
+        pointer_depth = 0
+        reference = False
+        if len(tokens) > 0:
+            pointer_depth = tokens.count("POINTER TO")
```

Comment:
Hmm, the IEC standard is case-insensitive. 

I think lark doesn't correct the case for us, so this may fail when the source contains `"pointer to"` instead of `"POINTER TO"`. Worth a test case, at the very least.

Reply:
well, I feel foolish for not thinking of the casing! I'm adding a little map-magic to format everything to uppercase and use those for the checks :smile: